kybra/compiler/kybra_generate/src/vm_value_conversion/try_from_vm_value_impls/numeric.rs

```diff
@@ -2,162 +2,260 @@ use proc_macro2::TokenStream;
                 }
             }
         }
 
-        impl CdkActTryFromVmValue<ic_cdk::export::candid::Int, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef {
-            fn try_from_vm_value(self, vm: &rustpython::vm::VirtualMachine) -> Result<ic_cdk::export::candid::Int, CdkActTryFromVmValueError> {
-                let int_result: Result<rustpython_vm::builtins::PyIntRef, _> = self.try_into_value(vm);
+        impl CdkActTryFromVmValue<ic_cdk::export::candid::Int, rustpython_vm::builtins::PyBaseExceptionRef, &rustpython::vm::VirtualMachine>
+            for rustpython::vm::PyObjectRef
+        {
+            fn try_from_vm_value(
+                self,
+                vm: &rustpython::vm::VirtualMachine,
+            ) -> Result<ic_cdk::export::candid::Int, rustpython_vm::builtins::PyBaseExceptionRef>
+            {
+                let int_result: Result<rustpython_vm::builtins::PyIntRef, _> =
+                    self.try_into_value(vm);
 
                 match int_result {
                     Ok(int) => Ok(ic_cdk::export::candid::Int(int.as_bigint().clone())),
-                    Err(_) => Err(CdkActTryFromVmValueError("TypeError: PyObjectRef is not a PyIntRef".to_string()))
+                    Err(_) => Err(vm.new_type_error("PyObjectRef is not a PyIntRef".to_string())),
                 }
             }
         }
 
-        impl CdkActTryFromVmValue<i128, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef {
-            fn try_from_vm_value(self, vm: &rustpython::vm::VirtualMachine) -> Result<i128, CdkActTryFromVmValueError> {
+        impl CdkActTryFromVmValue<i128, rustpython_vm::builtins::PyBaseExceptionRef, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef {
+            fn try_from_vm_value(
+                self,
+                vm: &rustpython::vm::VirtualMachine,
+            ) -> Result<i128, rustpython_vm::builtins::PyBaseExceptionRef> {
                 match self.try_into_value(vm) {
                     Ok(value) => Ok(value),
-                    Err(err) => Err(CdkActTryFromVmValueError("TypeError: Could not convert PyObjectRef to i128".to_string()))
+                    Err(err) => {
+                        Err(vm.new_type_error("Could not convert PyObjectRef to i128".to_string()))
+                    }
                 }
             }
         }
 
-        impl CdkActTryFromVmValue<i64, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef {
-            fn try_from_vm_value(self, vm: &rustpython::vm::VirtualMachine) -> Result<i64, CdkActTryFromVmValueError> {
+        impl CdkActTryFromVmValue<i64, rustpython_vm::builtins::PyBaseExceptionRef, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef {
+            fn try_from_vm_value(
+                self,
+                vm: &rustpython::vm::VirtualMachine,
+            ) -> Result<i64, rustpython_vm::builtins::PyBaseExceptionRef> {
                 match self.try_into_value(vm) {
```

Comment:
Check if try_into_value returns a PyBaseExceptionRef, if so we could collapse a lot of this

Reply:
Good catch. I've removed all these unnecessary matches and let the native exception bubble up. Taken care of with 1d53a56d9a46d5b97fd3f375ecef83af0dea6358